spezi-module/onboarding/src/main/java/edu/stanford/spezikt/spezi_module/onboarding/invitation/InvitationCodeScreen.kt

```diff
@@ -0,0 +1,88 @@
+package edu.stanford.spezikt.spezi_module.onboarding.invitation
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.hilt.navigation.compose.hiltViewModel
+import edu.stanford.spezikt.core.design.component.SpeziValidatedOutlinedTextField
+import edu.stanford.spezikt.core.design.theme.Colors.onPrimary
+import edu.stanford.spezikt.core.design.theme.Colors.primary
+import edu.stanford.spezikt.core.design.theme.Sizes
+import edu.stanford.spezikt.core.design.theme.Spacings
+import edu.stanford.spezikt.core.design.theme.TextStyles.titleLarge
+
+@Composable
+fun InvitationCodeScreen(
```

Comment:
Proposal for compose screens

For each screen, we should provide a parameterless composable, e.g. `InvitationCodeScreen()` and another composable `InvitationCodeScreen(uiState, actions, events)` where where actions and events can be present or not depending on the screen - in your case here not needed.

Inside the first composable, we will read the state, actions and events from the view model and call the other composable with the required parameters. The first composable can be used for integration tests, the second one can be used for ui tests :)

In your example here, we can just extract the content after line 35 in a new `InvitationCodeScreen(uiState, actions)` composable below in this file:

```kotlin
@Composable
fun InvitationCodeScreen() {
    val viewModel = hiltViewModel<InvitationCodeViewModel>()
    val uiState by viewModel.uiState.collectAsState()

    InvitationCodeScreen(uiState = uiState, onAction = viewModel::onAction)
}
```

Reply:
good idea. will also make previews easier :)